clicommand/agent_start.go

```diff
@@ -758,6 +759,18 @@ var AgentStartCommand = cli.Command{
 		))
 		defer done()
 
+		// Verify that the bootstrap buildkite-agent executable matches the current host agent
+		// executable. In development builds, it exits on mismatch; otherwise it only logs a warning.
+		// This is to avoid confusion when making changes to the buildkite-agent but forgetting to
+		// update the buildkite-agent binary available from $PATH
+		if err := checkBinaryPaths(); err != nil {
+			if version.IsDevelopmentBuild() {
+				return fmt.Errorf("check binary paths: %s", err)
```

Comment:
There might be situations where we want to allow this. I think it's good to err on the side of blowing up, rather than behaving unexpectedly, but I wonder if we should have a parameter to allow it?

Reply:
given that it's only used for development builds, i think if we need to turn it off for dev builds we can just comment out the code?